templates/company-board/test/company-board.js

```diff
@@ -0,0 +1,185 @@
+      await assertRevert(companyTemplate.newTokensAndInstance.request('id', [], SHARE_HOLDERS, SHARE_STAKES), 'COMPANY_MISSING_BOARD_MEMBERS')
+    })
+
+    it('reverts when the given holders and stakes length do not match', async () => {
+      await assertRevert(companyTemplate.newTokensAndInstance.request('id', BOARD_MEMBERS, [], SHARE_STAKES), 'COMPANY_INVALID_HOLDERS_STAKES_LEN')
+      await assertRevert(companyTemplate.newTokensAndInstance.request('id', BOARD_MEMBERS, SHARE_HOLDERS, []), 'COMPANY_INVALID_HOLDERS_STAKES_LEN')
+    })
+
+    it('reverts when there was no DAO cache', async () => {
+      await assertRevert(companyTemplate.setupInstance.request(BOARD_MEMBERS, SHARE_HOLDERS, SHARE_STAKES), 'COMPANY_MISSING_DAO_CACHE')
+    })
+  })
+
+  context('when the creation succeeds', () => {
+    before('create company with board entity', async () => {
+      daoID = `id-${Math.floor(Math.random() * 1000)}`
+      tokenReceipt = await companyTemplate.newTokens({ from: deployer })
+      prepareReceipt = await companyTemplate.prepareInstance(daoID, { from: deployer })
+      setupReceipt = await companyTemplate.setupInstance(BOARD_MEMBERS, SHARE_HOLDERS, SHARE_STAKES, { from: deployer })
+
+      dao = Kernel.at(getEventArgument(prepareReceipt, 'DeployDao', 'dao'))
+      boardToken = MiniMeToken.at(getEventArgument(tokenReceipt, 'DeployToken', 'token', 0))
+      shareToken = MiniMeToken.at(getEventArgument(tokenReceipt, 'DeployToken', 'token', 1))
+      acl = ACL.at(await dao.acl())
+    })
+
+    before('load apps', async () => {
+      const events = decodeEvents(setupReceipt.receipt, Kernel.abi, 'NewAppProxy')
+      const agentEvents = events.filter(e => e.args.appId === APP_IDS.agent)
+      const votingEvents = events.filter(e => e.args.appId === APP_IDS.voting)
+      const financeEvents = events.filter(e => e.args.appId === APP_IDS.finance)
+      const tokenManagerEvents = events.filter(e => e.args.appId === APP_IDS['token-manager'])
+
+      assert.equal(agentEvents.length, 1, 'should have deployed 1 agent app')
+      assert.equal(votingEvents.length, 2, 'should have deployed 2 voting apps')
+      assert.equal(financeEvents.length, 1, 'should have deployed 1 finance app')
+      assert.equal(tokenManagerEvents.length, 2, 'should have deployed 2 token manager apps')
+
+      agent = Agent.at(agentEvents[0].args.proxy)
+      finance = Finance.at(financeEvents[0].args.proxy)
+      boardVoting = Voting.at(votingEvents[0].args.proxy)
+      shareVoting = Voting.at(votingEvents[1].args.proxy)
+      boardTokenManager = TokenManager.at(tokenManagerEvents[0].args.proxy)
+      shareTokenManager = TokenManager.at(tokenManagerEvents[1].args.proxy)
+    })
+
+    it('costs less than 6e6 each tx', async () => {
+      assert.isAtMost(tokenReceipt.receipt.gasUsed, 3.5e6, 'tokens creation should cost almost 3.5e6 gas')
+      assert.isAtMost(prepareReceipt.receipt.gasUsed, 1.7e6, 'prepare DAO should cost almost 1.7e6 gas')
```

Comment:
I'm wondering if we can group the token creation and initial org creation into one step, so this template just takes two transactions to finish?